### ISSUE TITLE
Inline typed links are not being automatically reformatted.

### ISSUE THREAD

**Issue by fdt-dev** [zim-desktop-wiki/zim-desktop-wiki#1722]
**Describe the bug**
When writing Zim Wiki internal links inline like that:
`[[+A New Link]]`
which I always do when there is more than one word in the link
and I switch back and forth between the page containing this link and any other page
the newly typed link from above will not automatically be transformed into a clickable link that looks like that:
`+A New Link`

Hitting Ctrl-R will auto-convert the Wiki syntax.

**To Reproduce**
1. Go to any page in the current Wiki.
2. Type a new link like that: `[[+A New Link]]`
3. Switch to any other page
4. Switch back
5. See that the link still looks like `[[+A New Link]]`

**Expected behavior**
Zim Wiki versions previous to 0.74 auto converted the Wiki syntax when switching back and forth between pages.
It is worth noting that I would expect that Zim auto converts the Wiki syntax immediately after closing the brackets and entering an additional blank. That is what Zim Wiki is doing with most of the other Wiki syntax pieces like tags or bullet points and the likes.

**Screenshots**
![01 - Initial Page (before and after switching pages)](https://user-images.githubusercontent.com/35032197/140078754-19bd6562-2ee0-44b1-934c-fdbd6222006e.png)

![02 - Switching to another page](https://user-images.githubusercontent.com/35032197/140078829-0c414753-4f2f-42aa-914f-a8a4c20cf24d.png)

![03 - Expected look after switching back](https://user-images.githubusercontent.com/35032197/140078869-e31bf451-cc80-4b09-83dc-6acb08403061.png)

**Environment (please complete the following information):**
 - Version 0.74.2 (PPA)
 - Linux (possibly on Windows and mac os; not tested)
 - Language: German

**Comment by jaap-karssenberg** [comment]
Not a bug, simply never implemented. Open to adding this, but I believe hitting Ctrl-L is just as fast, so never saw much need for this. Also CamelCase links etc. are auto-formatted.

**Comment by fdt-dev** [comment]
Hello Jaap, I have to disagree. In my opinion it is a bug not to autoformat links like [[+New Link]].  Especially because pre 0.74.x versions did that correctly. And also: every other inline typed formatting is being auto-reformatted.

Yes, hitting Ctrl+L works, but why make it less convenient? That's slowing down the process. I love Zim precisely because it allows me to fast-type everything. Calling up a dialogue window is okay if the link is elswhere in the wiki and not a direct sub-page.

Switching away to a different page and switching back should reformat the link. Reason: That should be the equivalent of hitting Ctrl+R (I'm not asking for more, really, if you read my initial bug-report closely.) Everything else is being auto-reformatted. Please implement it.

I disabled CamelCase altogether, because I have to write CamelCase words which can not become links. That's not really a good argument.

Hitting Ctrl+R may not be that difficult, but is your argument, that every formatting should only become visible after hitting Ctrl+R? Because that'd be inconvenient.

If you don't want to do it, fine, it's your software. I just don't understand why something that was working fine in previous releases should not work anymore in a new and improved release.

**Comment by jaap-karssenberg** [comment]
No need to preach. All I was saying that as far as I can see this is not a
bug in the sense that it is a broken feature, but an new feature.

Just double checked the code, and can't find any reformatting code that
would pick up these kind of links. In what previous version do you believe
this did work ?

**Comment by fdt-dev** [comment]
I'm sorry, didn't mean to preach. And I probably wouldn't have written it that way if there were no discussions about UX in other bug-reports.

Like I said: it worked in releases pre 0.74.x
The way it worked was:
1. Type a link like this [[+New Link]]
2. select a different page
3. go back to the initial page
4. link is converted to a clickable link

I had to go back to a previous version of Zim, which is surprisingly difficult unless uninstalling any currently installed version. A VM with Xubuntu 18.04 helped. I was able to run Zim 0.73.5 directly from source.

Here are screenshots:

**Zim Version 0.73.5**
![Bildschirmfoto_2021-11-29_15-04-40](https://user-images.githubusercontent.com/35032197/143882402-d347cb4a-67a1-4873-bcfb-2c5e2586dbdf.png)

**Initial Page Setup**
![Bildschirmfoto_2021-11-29_15-04-47](https://user-images.githubusercontent.com/35032197/143882424-33588390-4530-4cb1-9bda-deff4c2d4ef7.png)

**Select Page "Test"**
![Bildschirmfoto_2021-11-29_15-04-56](https://user-images.githubusercontent.com/35032197/143882489-d17f97e7-f615-4a37-a043-d53c044ab5ba.png)

**Go Back to Home**
![Bildschirmfoto_2021-11-29_15-05-03](https://user-images.githubusercontent.com/35032197/143882509-9a6ce532-cb7a-474d-a82c-9bf421b9344c.png)


I did not press Ctrl-R at any point. I just went to a different page and back.


Note here: the link was not auto-converted while staying on the same page. I'm not assuming that. That's the important detail. I'm assuming that the page redraw did act as if Ctrl+R was pressed. That's what I would expect.
Since it worked previously ... it really isn't a "new" feature per se. But of course I won't argue the point anymore.

**Comment by jaap-karssenberg** [comment]
Ah, now I understand. Should have read more carefully. What happens is that
the bigger is no longer reloaded in navigation but stays in memory. The
advantage is speed and e.g. undo works even after navigating back and
forth. The reload effect as in your scenario was not designed as such but a
side effect of the reloading.

Still seems to me that best way forward is via auto-formatting while
typing. Reversing the reload behavior is not an option as it has clear
advantages.



Op ma 29 nov. 2021 15:12 schreef fdt-dev ***@***.***>:

**Comment by fdt-dev** [comment]
I see, that makes sense. Well, I will be able to learn to work around this.

But one thing to consider: Could you please take a look whether my other issue 1723 could somehow be related to the behaviour you describe? (I'm not setting a link to 1723 because it may not be related; it's hard to say from my end.)
My assumption was initially that there is a caching issue involved and what you're describing here sounds very much like it could have an effect similar to what I describe in 1723.

I do understand the need for speed. I like fast software.

**Comment by jaap-karssenberg** [comment]
Yes, 2nd part of #1723 is related to the same caching mechanism. Will look
into that one as well.
